src/jsutils/instanceOf.ts

```diff
@@ -9,7 +9,7 @@ import { inspect } from './inspect.js';
 export const instanceOf: (value: unknown, constructor: Constructor) => boolean =
   /* c8 ignore next 6 */
   // FIXME: https://github.com/graphql/graphql-js/issues/2317
-  globalThis.process != null && globalThis.process.env.NODE_ENV === 'production'
+  globalThis.process != null && globalThis[process.env.NODE_ENV] === 'production'
```

Comment:
This might "fix" the issue but introduces another one, this will get replaced with `globalThis['development' | 'production']` rather than the intended way where we analyse the NODE_ENV

Reply:
changed my PR

Reply:
That's not what I meant 😅 more that this is breaking off of the existing behavior as described in https://github.com/graphql/graphql-js/pull/3927

Starting a thread on Discord atm how we should go about this